
# Wooldridge Chapter 1 exercises

## Jerry Hong 

## C1 

* wage1.info() run this to see the variables you have or check the description on the web
* Find the average education level in the sample. What are the lowest and highest years of education?
* Find the average hourly wage in the sample. 
* 	How many women are in the sample? How many men?

In [2]:
# import necessary packages
import wooldridge as woo
import numpy as np
import pandas as pd

In [5]:
# finding the values 
wage1 = woo.dataWoo('wage1')
wage1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 24 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   wage      526 non-null    float64
 1   educ      526 non-null    int64  
 2   exper     526 non-null    int64  
 3   tenure    526 non-null    int64  
 4   nonwhite  526 non-null    int64  
 5   female    526 non-null    int64  
 6   married   526 non-null    int64  
 7   numdep    526 non-null    int64  
 8   smsa      526 non-null    int64  
 9   northcen  526 non-null    int64  
 10  south     526 non-null    int64  
 11  west      526 non-null    int64  
 12  construc  526 non-null    int64  
 13  ndurman   526 non-null    int64  
 14  trcommpu  526 non-null    int64  
 15  trade     526 non-null    int64  
 16  services  526 non-null    int64  
 17  profserv  526 non-null    int64  
 18  profocc   526 non-null    int64  
 19  clerocc   526 non-null    int64  
 20  servocc   526 non-null    int64 

In [37]:
# setup 
wage1 = woo.dataWoo('wage1')


eductable = wage1['educ'].describe()
avewage = np.mean(wage1.wage)
femalecount = wage1['female'].value_counts()[1]
malecount =  wage1['female'].value_counts()[0]
meaneduc = wage1['educ'].mean()
minyears = wage1['educ'].min()
maxyears = wage1['educ'].max()

stats_new = {
    'Values': [avewage, femalecount, malecount],
    'Stats': ['Average wage', 'female', 'male']
}

stats_table = pd.DataFrame(stats_new).set_index('Stats')

print(f'Educ Stats \n{eductable}\n')

print(round(stats_table,2))

#stats = pd.DataFrame()

# CPI-1982 100 # CPI-1976 = 56.9 & CPI_2013= 233.0
CPIratio76_13 = 233/56.9
wage13 = wage1['wage']*CPIratio76_13
avewage13 = np.mean(wage13)

print(f'\nAverage wage in 2013; {avewage13}\n')
print(f'Mean education level; {meaneduc}')
print(f'Lowest years of education; {minyears}')
print(f'Highest years of education; {maxyears}')


Educ Stats 
count    526.000000
mean      12.562738
std        2.769022
min        0.000000
25%       12.000000
50%       12.000000
75%       14.000000
max       18.000000
Name: educ, dtype: float64

              Values
Stats               
Average wage     5.9
female         252.0
male           274.0

Average wage in 2013; 24.143970531201788

Mean education level; 12.562737642585551
Lowest years of education; 0
Highest years of education; 18


## C3 Computer exercise three

The data in MEAP01 are for the state of Michigan in the year 2001. Use these data to answer the following questions.

(i). Find the largest and smallest values of math4. Does the range make sense? Explain.

(ii). How many schools have a perfect pass rate on the math test? What percentage is this of the total sample?

(iii). How many schools have math pass rates of exactly 50%?

(iv). Compare the average pass rates for the math and reading scores. Which test is harder to pass?

(v). Find the correlation between math4 and read4. What do you conclude?

(vi). The variable exppp is expenditure per pupil. Find the average of exppp along with its standard deviation. Would you say there is wide variation in per pupil spending?

In [38]:
# Load dataset meap01

meap01 = woo.dataWoo('meap01')
meap01.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1823 entries, 0 to 1822
Data columns (total 11 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   dcode    1823 non-null   float64
 1   bcode    1823 non-null   int64  
 2   math4    1823 non-null   float64
 3   read4    1823 non-null   float64
 4   lunch    1823 non-null   float64
 5   enroll   1823 non-null   int64  
 6   expend   1823 non-null   float64
 7   exppp    1823 non-null   float64
 8   lenroll  1823 non-null   float64
 9   lexpend  1823 non-null   float64
 10  lexppp   1823 non-null   float64
dtypes: float64(9), int64(2)
memory usage: 156.8 KB


In [47]:
# calculate the values
tablestats = meap01.describe()

meap01[['math4', 'read4', 'exppp']].describe()


print(round(tablestats,2))
perfmath4 = meap01[meap01['math4']==100]['math4'].count()
midmath4 = meap01[meap01['math4']==50]['math4'].count()
perfpropmath4 = perfmath4/meap01['math4'].count()

corr = meap01['math4'].corr(meap01['read4'])
print(round(corr,2))
print(f'Schools with perfect passing rate (math4); {perfmath4}')
print(f'Schools with 50% passing rate (math4); {midmath4}')
print(f'Proprotion of schools with perfect passing rate (math4); {perfpropmath4}')



          dcode    bcode    math4    read4    lunch   enroll      expend  \
count   1823.00  1823.00  1823.00  1823.00  1823.00  1823.00     1823.00   
mean   53028.27  3394.56    71.91    60.06    39.25   401.94  2036983.94   
std    23907.06  2442.93    19.95    19.15    26.42   169.83   864936.06   
min     1010.00     1.00     0.00     0.00     0.00    62.00   275985.00   
25%    33901.50  1385.50    61.60    48.90    16.61   287.00  1474465.00   
50%    55100.00  2794.00    76.40    62.70    35.10   379.00  1894043.00   
75%    77010.00  5084.50    87.00    73.90    59.07   484.00  2407636.50   
max    83070.00  8838.00   100.00   100.00   100.00  1496.00  7665998.00   

          exppp  lenroll  lexpend   lexppp  
count   1823.00  1823.00  1823.00  1823.00  
mean    5194.87     5.91    14.44     8.53  
std     1091.89     0.42     0.41     0.21  
min     1206.88     4.13    12.53     7.10  
25%     4501.54     5.66    14.20     8.41  
50%     5078.25     5.94    14.45     8.53  


## Observations and Analysis

The range of the math4 values is 0 to 100. My initial thought is that this range is plausible because you can have either all or none of the students be satisfactory. There are 38 schools that have a perfect pass rate in 4th grade math, which makes up roughly 2% of the total sample size. There are 17 schools that have a pass rate of 50%. The average pass rates for math and reading are about 72% and 60% respectively, suggesting that the reading test is harder to pass. The correlation between math4 and read4 is 0.84. This suggests there is a relatively strong positive relationship between the two tests, indicating students who pass the math test are also likely to pass the reading test. From the table, the average expenditure per pupil (exppp) is 5194.87 with a standard deviation of 1091.89. This value to me suggests there is quite a wide variation in per pupil spending, especially considering the range between the min and max. 

